src/lib/account-settings.ts

```diff
@@ -181,3 +183,70 @@ export const uploadAvatarToWNFS = async (image: File): Promise<void> => {
     console.error(error)
   }
 }
+
+export const generateRecoveryKit = async (): Promise<string> => {
+  const {
+    program: {
+      components: { crypto, reference, storage }
+    },
+    username: {
+      full,
+      trimmed
+    }
+  } = getStore(sessionStore)
+
+  // Get the user's read-key and base64 encode it
+  const accountDID = await rootDID({ crypto, reference, storage })
```

Comment:
I would avoid using functions directly from component implementations.
I would use the following instead:
```ts
await reference.didRoot.lookup(username)
```
We could add a shorthand function for this to the program if that's easier.
Like we just did with `program.agentDID()`

---

Also, I'm thinking of removing that `rootDID` function from the wnfs-auth implementation.
It looks up the root DID via UCAN proofs, which we should probably avoid.
For two reasons:
1. This lookup should be consistent with the other lookups in Webnative. The other day I had a situation where the root DID from the UCAN didn't match the one in DNS, so I couldn't load my file system.
2. When we adopt UCAN 0.9, we won't have nested UCANs in the current sense anymore, they'll be CIDs or another type of pointer pointing at the other UCAN. So this recursive lookup might not be as easy anymore.

Thoughts? cc @bgins

Reply:
sounds good! i'll use `await reference.didRoot.lookup(username)` for now :ok_hand:

Reply:
updated 👍🏼 maybe we can look at adding the shorthand function to webnative in an upcoming release?

Reply:
Sounds good 👍

Reply:
Continueing root DID discussion here: https://github.com/fission-codes/webnative/pull/435/files#r1047523803

Reply:
> Also, I'm thinking of removing that rootDID function from the wnfs-auth implementation.
It looks up the root DID via UCAN proofs, which we should probably avoid.
For two reasons:
> 1. This lookup should be consistent with the other lookups in Webnative. The other day I had a situation where the root DID from the UCAN didn't match the one in DNS, so I couldn't load my file system.
> 2. When we adopt UCAN 0.9, we won't have nested UCANs in the current sense anymore, they'll be CIDs or another type of pointer pointing at the other UCAN. So this recursive lookup might not be as easy anymore.

> Thoughts? cc @bgins

A bit late to the party on this, but I'm agreed that we probably won't want to rely on finding the root DID from the UCAN in the future. 👌